src/librbd/io/SimpleSchedulerObjectDispatch.cc

```diff
@@ -108,6 +148,18 @@ bool SimpleSchedulerObjectDispatch<I>::write(
   }
 
   dispatch_delayed_requests(object_no, on_finish);
+
+  if (m_latency_stats) {
```

Comment:
Could this new ```FunctionContext``` be combined w/ the one created in ```dispatch_delayed_requests```? The latency seems like it should actually be calculated from the dispatch time anyway.

Reply:
@dillaman this could be moved to `dispatch_delayed_requests`. The reason why I did this in `write` method, because I wanted latency statistics collected only for `write` operations. Do you think there is no need in this distinction? Or if it is needed should I pass a yet another param to `dispatch_delayed_requests` to tell if it should collect latency?

And   I am not sure I understand this:

> The latency seems like it should actually be calculated from the dispatch time anyway.

Do you think there is something wrong with the current code? I am calculating the latency only for non-merged write requests. I thought it would be enough for statistics. Do you prefer we collect stats for merged (re-dispatched) requests too?

Reply:
I was trying to avoid another allocation / wrapped callback on the IO path if it wasn't needed. Unless you take a histogram by IO size approach, I think all writes should be tracked for latency -- not just non-merged writes. That means that delayed dispatch requests should start their latency timer from the point at which they are sent down to the next layer.